ide/css.editor/src/org/netbeans/modules/css/editor/module/spi/CssCompletionItem.java

```diff
@@ -609,6 +620,42 @@ public ElementKind getKind() {
         public int getSortPrioOverride() {
             return super.getSortPrioOverride() - (related ? 1 : 0);
         }
+
+        /**
+         * Escape the input for usage in CSS files.
+         *
+         * @param input
+         * @return
+         */
+        private static String escape(String input) {
```

Comment:
shouldn't NB use some encoding library, e.g. "OWASP Java Encoder" is fairly popular (and tiny).
https://github.com/OWASP/owasp-java-encoder
https://javadoc.io/static/org.owasp.encoder/encoder/1.2.3/org/owasp/encoder/Encode.html

Reply:
I would agree if we would talk about a web application, but in this case the usage is not security, but compatibility. 

I saw one thing in the description of the CSS escaper of owasp, that looks like a potential problem for my approach. I overlooked that the hexadecimal encoding and decoding needs a bounding whitespace or can have it. I'll revisit this and the CSS grammar.